core/src/main/java/org/infinispan/stream/CacheCollectors.java

```diff
@@ -0,0 +1,106 @@
+package org.infinispan.stream;
+
+import org.infinispan.commons.marshall.Externalizer;
+import org.infinispan.commons.marshall.SerializeWith;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+/**
+ * Helper class designed to be used to create a serializable Collector for use with
+ * {@link org.infinispan.CacheStream#collect(Collector)} from a supplier of a collector.  The problem is that the
+ * standard {@link java.util.stream.Collectors} class doesn't provide Serializable Collectors and no way to extend
+ * their functionality, so this class is used instead.
```

Comment:
Good point I forgot to override the method to add the comment :+1:

They can pass a non serializable collector however it must be on a collect method after a stateful intermediate operation to force entries local.